Intercom v2 source updates (#1012)
rudderlabs/rudderstack-docs: docs/sources/extract/intercom.mdx

- Authenticate RudderStack with Intercom by clicking the **Connect with Intercom** button.
- **Start Date**: Enter the date from which RudderStack should ingest the Intercom data. RudderStack will not replicate any data before this date.

### Destination settings

The following settings specify how RudderStack sends the data ingested from Intercom to the connected warehouse destination:

- **Table prefix**: RudderStack uses this prefix to create a table in your data warehouse and loads all your Intercom data into it.
- **Schedule Settings**: RudderStack gives you three options to ingest the data from Intercom:
    - **Basic**: Runs the syncs at the specified time interval. 
    - **CRON**: Runs the syncs based on the user-defined CRON expression.
    - **Manual**: You are required to run the syncs manually.

<div class="infoBlock">
For more information on the schedule types, refer to the <Link to="/sources/extract/common-settings/">Common Settings</Link> guide.
</div>

### Selecting the data to import

You can choose the Intercom data that you want to ingest by selecting the required resources.

> <img src="../../assets/cloud-extract-sources/intercomv2-5.png" alt="Selecting the data to import" />

Reviewer: If we can update this image, the current one appears quite blurry

Author: Done!

The below table mentions the sync types supported by the Intercom resources while syncing data to your warehouse destination:

| Resource | Sync type | Endpoint | Primary key |
| :---| :--- | :----| :----|
| `admins` | Full Refresh | `/admins` | `id` |
| `companies` | Incremental |  `/companies/scroll` | `id` |
| `company segments` | Incremental | `/companies/<id>/segments` | `id` |
| `company attributes` | Full Refresh | `/data_attributes?model=company` | `name` |
| `conversations` | Incremental | `/conversations` | `id` |
| `conversation parts` |  Incremental | `/conversations/<id>` | `id` |
| `contacts` | Incremental | `/contacts` | `id` |
| `contact attributes` |Full Refresh | `/data_attributes?model=contact` | `name` |
| `segments` | Incremental | `/segments` | `id` |
| `tags` | Full Refresh | `/tags` | `name` |
| `teams` | Full Refresh | `/teams` | `name` |

<div class="infoBlock">
For more information on the <strong>Full Refresh</strong> and <strong>Incremental</strong> sync modes, refer to the <Link to="/sources/extract/common-settings/#sync-modes">Common Settings</Link> guide.
</div>

<div class="warningBlock">

> Any resource that supports the <strong>Incremental</strong> sync type also supports <strong>Full Refresh</strong>, by default. However, to sync your data using the <strong>Full Refresh</strong> mode, you need to <a href="mailto:support@rudderstack.com">contact the RudderStack team</a>.

Reviewer: ```suggestion
Any resource that supports the <strong>Incremental</strong> sync mode also supports <strong>Full Refresh</strong>, by default. To sync your data using the <strong>Full Refresh</strong> mode in such cases, you need to <a href="mailto:support@rudderstack.com">contact the RudderStack team</a>.
```

</div>

Intercom is now configured as a source. RudderStack will start ingesting data from Intercom as per your specified schedule and frequency.

You can further connect this source to your data warehouse by clicking the **Add Destination** button, as shown:

> <img src="../../assets/cloud-extract-sources/intercomv2-6.png" alt="Configuring the source" />

Author: ```suggestion
<img src="../../assets/cloud-extract-sources/add-destination.png" alt="Adding a destination" />
```

### Destination settings

The following settings specify how RudderStack sends the data ingested from Intercom to the connected warehouse destination:

- **Table prefix**: RudderStack uses this prefix to create a table in your data warehouse and loads all your Intercom data into it.
- **Schedule Settings**: RudderStack gives you three options to ingest the data from Intercom:
    - **Basic**: Runs the syncs at the specified time interval. 
    - **CRON**: Runs the syncs based on the user-defined CRON expression.
    - **Manual**: You are required to run the syncs manually.

<div class="infoBlock">
For more information on the schedule types, refer to the <Link to="/sources/extract/common-settings/">Common Settings</Link> guide.
</div>

### Selecting the data to import

You can choose the Intercom data that you want to ingest by selecting the required resources.

<img src="../../assets/cloud-extract-sources/intercomv2-5.png" alt="Selecting the data to import" />

The below table mentions the sync types supported by the Intercom resources while syncing data to your warehouse destination:

> | Resource | Full Refresh sync | Incremental sync | 

Author: @a-rampalli could you please share the list of associated endpoints and the primary key details for the below resources? cc @nidhilashkari17
